### PR TITLE
Hide IAP debug upgrades behind experimental toggle except for debug monthly

### LINKED ISSUE #9894: [IAP] Hide Upgrades behind experimental toggle
From https://github.com/woocommerce/woocommerce-ios/pull/9874/files#r1221253705

We already have an experimental toggle for testing IAP, we could use the same one for hiding/showing Upgrades logic.

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #9894  . Branched from https://github.com/woocommerce/woocommerce-ios/pull/9931 , that PR needs to be merged first.

## Description
This PR hides all debug IAP upgrades (except for essential debug monthly) behind the IAP experimental toggle. With this change we will only see one debug plan on debug builds if the toggle is off, and see all available debug plans on debug builds if the toggle is on, so we close the gap with the intended future production behaviour for testing purposes.

## Testing instructions
On a Free Trial store, where you're the owner or administrator:
* Go to Menu > Settings > Experimental Features > Turn off the In-App Purchases toggle
* Go to My Store > Tap "Upgrade Now" > See only 1 debug plan available for purchase
* Go to Menu > Settings > Experimental Features > Turn on the In-App Purchases toggle
* Go to My Store > Tap "Upgrade Now" > See multiple debug plans available for purchase

## Screenshots
| Experimental toggle OFF | Experimental toggle ON |
|--------|--------|
| Simulator Screen Shot - iPhone 11 Pro - 2023-06-13 at 12 03 30 | Simulator Screen Shot - iPhone 11 Pro - 2023-06-13 at 12 03 46 |

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift [hunk 1]
```diff
@@ -38,6 +38,10 @@ struct UpgradesView: View {
         ServiceLocator.stores.sessionManager.defaultSite?.name
     }
 
+    private var showingInAppPurchasesDebug: Bool {
+        ServiceLocator.generalAppSettings.betaFeatureEnabled(.inAppPurchases)
+    }
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift:43]: I understand the logic for using the existing "Debug In-App Purchases" flag, as it reveals all the debug plans.

However, another way we could think of it is that this is an M2 flag, to start showing all the other plans as options. That way we can use it in future to switch the plan details UI out.

Doesn't need to change now, it's just a thought exercise – wdyt? Also, an M2 flag should probably enable only the 4 plans that we're going to have in M2, not all plans. E.g. we wouldn't want it to show multi-site plans if we add extra subscription groups.

**Review by joshheald — APPROVED** [review]
Works as described... one suggestion/thought added

File: WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift:43]: Thanks for the review! That makes sense, I'm wondering if the right moment to do this it's when we add the initial real plan, let's say:
- M1 flag: Shows real Essential monthly when experimental toggle is off, shows debug monthly when the toggle is on.
- M2 flag: Shows real additional plans (+3) when experimental toggle is off (would show no real plans for the moment), and the debug ones with the toggle on.

The experimental toggle just switches plans used on M1 and M2 between real and debug. This way the different batches of plans we release would still be hidden behind their individual feature flags, but we can toggle the switch for testing in developer builds as needed. Does that make sense?

Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift:43]: Logged it here: https://github.com/woocommerce/woocommerce-ios/issues/9948

**Pull request merged**